packages/babel-parser/src/parser/expression.js

```diff
@@ -747,6 +742,77 @@ export default class ExpressionParser extends LValParser {
     }
   }
 
+  isSliceSimpleArgument(expr: N.Expression): boolean {
+    const { type, extra } = expr;
+    return (
+      (type === "NumericLiteral" || type === "Identifier") &&
+      extra?.parenthesized !== true
+    );
+  }
+
+  checkSliceArgument(pos: number, expr: ?N.Expression) {
+    if (
+      !expr ||
+      this.isSliceSimpleArgument(expr) ||
+      (expr.type === "UnaryExpression" &&
+        expr.operator === "-" &&
+        expr.extra?.parenthesized !== true &&
+        this.isSliceSimpleArgument(expr.argument))
```

Comment:
I don't think `-foo` is allowed, but `-Infinity` should be allowed, although they are syntactically equivalent.

Reply:
I don't think that `-Infinity` is valid, since it is currently restricted to `Identifiers` and `DecimalDigits` (https://github.com/tc39/proposal-slice-notation#can-the-upper-bound-lower-bound-or-the-step-argument-be-an-arbitrary-expression).
Althought, the readme itself contracdits it by using negative numbers. Also, there is https://github.com/tc39/proposal-slice-notation/issues/16 which didn't receive a response.

I am ok either with the current status of this PR, or with making it more restrictive and only allow `- NumericLiteral`. Since this is a stage 1 proposal, it's expected that we will fix it when the semantics are clearer.

Reply:
If `- NumericLiteral` is allowed, it is strange that `- Infinity` is not allowed given that `Infinity` is allowed.

I tend to making it restrictive without the loss of expressive power. Since `- Infinity` could not be expressed further simple, I suggest we support it unless the proposal explicitly bans `-Infinity`.